src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3021,6 +3021,11 @@ object Types {
 
   object ErrorType extends ErrorType
 
+  /* Type used to track Select nodes that could not resolve a member and their qualifier is a scala.Dynamic. */
+  abstract class TryDynamicCallType extends UncachedGroundType with ValueType
+
+  object TryDynamicCallType extends TryDynamicCallType
+
```

Comment:
The idea is that it behaves like `ErrorType` on any operation until it is caught and handled. I considered making it a subtype of `ErrorType` but this seamed a bit dangerous and with this alternative there was only one place where I had to add it to a `case` to do the same as `ErrorType`.